### PR TITLE
iss141 Create summary service and transform a query to a correct mongoose format

### PR DESCRIPTION
**Pull request by parnurzeal**
PR for #141  to create Restful resource for /summaries 
It is meant for storing the daily summary. The total summary of range period will then be managed by another routing called `/total_summary`

Note that we also need to create `trigger` method to recalculate and fill in each day summary. I will do it in the next PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "pug": "^2.0.0-beta4",
     "serve-favicon": "^2.3.0",
     "superagent": "^2.0.0",
+    "underscore": "^1.8.3",
```
Comment (supasate) [inline package.json]: Is lodash better?

File: src/services/summary/hooks/modify-search-query.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  if (startDate) {
+    date.$gte = startDate;
+  }
+  if (endDate) {
+    date.$lte = endDate;
+  }
+  if (!_.isEmpty(date)) {
+    hook.params.query.date = date; // eslint-disable-line no-param-reassign
+  }
+  console.log(hook.params);
```
Comment (supasate) [inline src/services/summary/hooks/modify-search-query.js]: remove console.log

File: test/services/summary/index.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const assert = require('assert');
```
Comment (supasate) [inline test/services/summary/index.test.js]: Use expect from test-helper

File: test/services/summary/index.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+const app = require('../../../src/app');
+
+describe('summary service', () => {
+  it('registered the summaries service', () => {
+    assert.ok(app.service('summaries'));
```
Comment (supasate) [inline test/services/summary/index.test.js]: `expect(app.service('summaries')).to.be.ok();`

File: src/services/summary/summary-model.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const Schema = mongoose.Schema;
+
+const byDepartmentSchema = new Schema({
+  department: { type: Schema.Types.ObjectId, ref: 'Department', required: true },
+  assigned: { type: Number },
+  processing: { type: Number },
+  resolved: { type: Number },
+});
+
+const summarySchema = new Schema({
```
Comment (supasate) [inline src/services/summary/summary-model.js:12]: Do we need to update this document every time we change pins' states?

Comment (parnurzeal) [reply src/services/summary/summary-model.js:12]: My plan is to use activity log to recalculate the number of each day. It won't be real-time for now. You would have to trigger it through `triggter` routing that I am gonna make. WDYT?

File: test/services/summary/index.test.js (hunk 4 above)
Comment (parnurzeal) [reply test/services/summary/index.test.js]: Done.

File: test/services/summary/index.test.js (hunk 3 above)
Comment (parnurzeal) [reply test/services/summary/index.test.js]: Done.

File: src/services/summary/hooks/modify-search-query.js (hunk 2 above)
Comment (parnurzeal) [reply src/services/summary/hooks/modify-search-query.js]: Done.

File: package.json (hunk 1 above)
Comment (parnurzeal) [reply package.json]: Done.

File: src/services/summary/summary-model.js (hunk 5 above)
Comment (supasate) [reply src/services/summary/summary-model.js:12]: Ok. Let's try that way.

File: src/services/summary/hooks/modify-search-query.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-const _ = require('underscore');
+const _ = require('loadash');
```
Comment (supasate) [inline src/services/summary/hooks/modify-search-query.js]: lodash

File: package.json [hunk 7]
```diff
@@ -38,6 +38,7 @@
     "feathers-rest": "^1.4.2",
     "feathers-socketio": "^1.4.1",
     "google-cloud": "^0.44.0",
+    "loadash": "0.0.1",
```
Comment (supasate) [inline package.json]: no need. we already have lodash.

Comment (parnurzeal) [reply package.json]: oh already?? from where?

Comment (supasate) [reply package.json]: I found it in swap lat long hook.

**Pull request merged**